src/router/legacyPermalinkManagement.routerPlugin.js

```diff
@@ -94,26 +78,26 @@ const handleLegacyParam = (
         // taking all layers related param aside so that they can be processed later (see below)
         // this only occurs if the syntax is recognized as a mf-geoadmin3 syntax (or legacy)
         case 'layers':
-            if (isLayersUrlParamLegacy(legacyValue)) {
-                // for legacy layers param, we need to give the whole search query
-                // as it needs to look for layers, layers_visibility, layers_opacity and
-                // layers_timestamp param altogether
-                const layers = getLayersFromLegacyUrlParams(
-                    store.state.layers.config,
-                    window.location.search
-                )
-                newValue = layers.map((layer) => transformLayerIntoUrlString(layer)).join(';')
-                log.debug('Importing legacy layers as', newValue)
-            } else {
-                // if not legacy, we let it go as it is
-                newValue = legacyValue
-            }
+            // for legacy layers param, we need to give the whole search query
+            // as it needs to look for layers, layers_visibility, layers_opacity and
+            // layers_timestamp param altogether
```

Comment:
Comment not 100% correct you don't pass the whole query but only the part that are needed